### PR TITLE
[SG-33093.8] Enable accessibilityAudit on `client/web/src/integration/search-contexts.test.ts`

### PR DESCRIPTION
**Pull request by gitstart-sourcegraph**
## Description

Enable accessibility audit on `client/web/src/integration/search-contexts.test.ts`

## Implementation details
Here's how we should implement:

-  Add `accessibilityAudit` in the same place where we call `percySnapshot` in the integration tests.
- Run the tests locally and observe any failures..

To fix the failures, we should:

1. If it's simple or quick to fix, then make the changes on the branch. There should be relevant documentation attached to the test error that will help you work out how to fix this. 
2. If it's complex, or will take a long time to fix. Add the `a11y-ignore` class to exclude the element from the audit, raise an issue specifically to solve this problem, leave a comment next to the class with a link to the issue.

## Refs

[SG Issue](https://github.com/sourcegraph/sourcegraph/issues/33093)
[Gitstart Ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-33093)
## Test plan

Run the integration test `ENTERPRISE=1 HEADLESS=true yarn test-integration:base client/web/src/integration/search-contexts.test.ts
`
## Success criteria

- All integration tests with Percy snapshots have `accessibility audit` enabled

## App preview:

- [Web](https://sg-web-contractors-sg-33093-8.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-pvcfhnxmmg.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/search-ui/src/input/SearchContextDropdown.tsx [hunk 1]
```diff
@@ -136,7 +141,11 @@ export const SearchContextDropdown: React.FunctionComponent<SearchContextDropdow
                     )}
                 </code>
             </DropdownToggle>
-            <DropdownMenu positionFixed={true} className={styles.menu}>
+            {/*
+               a11y-ignore
+               Rule: "aria-required-children" (Certain ARIA roles must contain particular children)
```
Comment (limitedmage) [inline client/search-ui/src/input/SearchContextDropdown.tsx:146]: Can we file a work item specifically for this dropdown?

**Review by limitedmage — COMMENTED** [review]
Looks good so far, I'll take a second look when this is marked at ready to review.

File: client/search-ui/src/input/SearchContextDropdown.tsx (hunk 1 above)
Comment (gitstart-sourcegraph) [reply client/search-ui/src/input/SearchContextDropdown.tsx:146]: Created an issue for this here #34348

File: client/web/src/search/panels/SavedSearchesPanel.tsx [hunk 2]
```diff
@@ -99,7 +99,7 @@ export const SavedSearchesPanel: React.FunctionComponent<Props> = ({
                 {savedSearches
                     ?.filter(search => (showAllSearches ? true : search.namespace.id === authenticatedUser?.id))
                     .map(search => (
-                        <li key={search.id} className="text-monospace test-saved-search-entry mb-2">
+                        <li key={search.id} className="text-monospace test-saved-search-entry d-block mb-2">
```
Comment (limitedmage) [inline client/web/src/search/panels/SavedSearchesPanel.tsx:102]: Is `display: block` not the default already for `li`?

**Review by limitedmage — APPROVED** [review]

File: client/web/src/search/panels/SavedSearchesPanel.tsx (hunk 2 above)
Comment (gitstart-sourcegraph) [reply client/web/src/search/panels/SavedSearchesPanel.tsx:102]: @limitedmage No, it's not. the default value for it is `list-item` and it causes some UI changes.

**Review by umpox — APPROVED** [review]
Thanks!

**Pull request merged**